lib/src/r_tree/r_tree.dart

```diff
@@ -18,8 +18,8 @@ part of r_tree;
 
 /// A two dimensional index of data that allows querying by rectangular areas
 class RTree<E> {
-  Node<E> _root;
-  int _branchFactor;
+  late Node<E> _root;
+  int? _branchFactor;
```

Comment:
```suggestion
  int _branchFactor = 16;
  late Node<E> _root = LeafNode<E>(_branchFactor);
```
I'd like to see branchFactor be non-nullable and default to 16 like in the constructor. Then all the leaf types can also have non-nullable branchFactors as well.